torch/utils/benchmark/utils/compile.py

```diff
@@ -0,0 +1,192 @@
+import torch
+
+__all__ = ["bench_all", "benchmark_compile"]
+
+import torch
+import torch._dynamo as torchdynamo
+import time
+import warnings
+from typing import Optional, List
+
+# Added this otherwise all the warnings from each backend will be printed
+# Not sure if this warning will propgate if user is not using this config
+warnings.filterwarnings("ignore")
+
+
+
+class SimpleTable:
+    """
+    This is a simple table class that can be used to print a table of data.
+    The primary reason it's here is to avoid taking an external dependency.
+    If it's useful, it can be moved to a more general location.
+    """
+    def __init__(self, field_names : List[str]):
+        self.field_names = field_names
+        self.rows = []
+
+    def add_row(self, row):
+        self.rows.append(row)
+
+    def __str__(self):
+        if not self.field_names or not self.rows:
+            return ""
+
+        formatted_rows = []
+        for row in self.rows:
+            formatted_row = []
+            for cell in row:
+                if isinstance(cell, float):
+                    formatted_cell = f"{cell:10.4f}"
+                else:
+                    formatted_cell = str(cell)
+                formatted_row.append(formatted_cell)
+            formatted_rows.append(formatted_row)
+
+        # Find the maximum width for each column
+        column_widths = [max([len(cell) for cell in column] + [len(header)]) for header, column in zip(self.field_names, zip(*formatted_rows))]
+
+        header = " | ".join([f"{str(field_name).ljust(width)}" for field_name, width in zip(self.field_names, column_widths)])
+        separator = "-+-".join(["-" * width for width in column_widths])
+
+        rows = []
+        for row in formatted_rows:
+            rows.append(" | ".join([f"{cell.ljust(width)}" for cell, width in zip(row, column_widths)]))
+
+        return f"{header}\n{separator}\n" + "\n".join(rows)
+
+def bench_loop(model : torch.nn.Module, sample_input : torch.Tensor, num_iters : int, is_training : bool =False, optimizer : torch.optim.Optimizer = None):
+    """
+    This is a simple loop that can be used to benchmark a model for either training or inference
+    It takes care of taking several measurements and averaging them
+    It also takes care of calling `cuda.synchronize()` if the model is on GPU
+    """
+    durations = []
+    for _ in range(num_iters):
+        start = time.time()
+        
+        if is_training and optimizer:
```

Comment:
is_training does nothing unless you pass optimizer; in that case, is there any point in an is_training parameter?